xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -1195,23 +1193,25 @@ bool CPVRClients::OpenStream(const CPVRChannel &tag, bool bIsSwitchingChannel)
   return bReturn;
 }
 
-bool CPVRClients::OpenStream(const CPVRRecording &tag)
+bool CPVRClients::OpenStream(const CPVRRecordingPtr &tag)
 {
   bool bReturn(false);
   CloseStream();
 
-  /* try to open the recording stream on the client */
-  PVR_CLIENT client;
-  if (GetConnectedClient(tag.m_iClientId, client) &&
-      client->OpenStream(tag))
+  if (tag)
```

Comment:
either return directly before CloseStream() or assert (cause programmer error), so this doesn't need another scope